File: FragmentManager/MainActivity.java
package com.example.myapplication;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import android.os.Bundle;
import android.view.View;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //  Add fragment
        getSupportFragmentManager()
                .beginTransaction()
                .add(R.id.frame, FirstFragment.newInstance("Hello", 1))
                .commit();
        //  Call method in fragment
        init();
    }

    private void init() {
        FragmentManager fm = getSupportFragmentManager();
        List<Fragment> fragments = fm.getFragments();
        for (Fragment fragment: fragments) {
            if (fragment instanceof FirstFragment) {
                //  Your example code
                ((FirstFragment)fragment).onBegin();
            }
        }
        //  Or get by index
        Fragment lastFragment = fragments.get(fragments.size() - 1);
    }
}

File: FragmentManager/MainActivity.kt
package com.example.myapplication

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class MainActivity: AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        //  Add fragment
        supportFragmentManager
                .beginTransaction()
                .add(R.id.frame, FirstFragment.newInstance("Hello", 1))
                .commit()
        //  Call method in fragment
        init()
    }
    
    private fun init() {
        val fm = supportFragmentManager
        val fragments = fm.fragments
        for (fragment in fragments) {
            if (fragment is FirstFragment) {
                //  Your example code
                (fragment as FirstFragment).onBegin()
            }
        }
        //  Or get by index
        val lastFragment = fragments[fragments.size - 1]
    }
}
